test/IntegrationTests/MinimalApiTests.cs

```diff
@@ -0,0 +1,69 @@
+// <copyright file="MinimalApiTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+using IntegrationTests.Helpers;
+using OpenTelemetry.Proto.Logs.V1;
+using Xunit.Abstractions;
+
+namespace IntegrationTests;
+
+public class MinimalApiTests : TestHelper
+{
+    public MinimalApiTests(ITestOutputHelper output)
+        : base("MinimalApi", output)
```

Comment:
Can you explain why is it preferred to adjust `TestApplication.Razor` instead of using the new test application?
I'd prefer to keep this new test application, as it is minimal example that reproduced the issue.

Reply:
According to me both `TestApplication.Razor` and `MinimalApiTests` are same, we are creating duplicate test apps here. Its better to modify the Razor test with the implementation added here.

Reply:
`TestApplication.Razor`, compared to `TestApplication.MinimalApi`, has additional configuration/services added related to razor pages, which are not relevant as far as issue is concerned.
What do you think about removing `RazorTests` and `TestApplication.Razor`?

Reply:
Yes, keep either of them. No strong preference.

Reply:
Removed `RazorTests` and `TestApplication.Razor` in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3003/commits/8451ff45c3b3fb9279d199091d647105f2a4a086
I also added test application created with worker template (uses generic host).